wandb/sdk/internal/internal.py

```diff
@@ -174,9 +191,20 @@ def filter(self, record):
 class HandlerThread(internal_util.RecordLoopThread):
     """Read records from queue and dispatch to handler routines."""
 
+    _record_q: "Queue[Record]"
+    _result_q: "Queue[Result]"
+    _stopped: "Event"
```

Comment:
I could be mistaken here, but I think the quoted, [forward reference](https://docs.python.org/3/library/typing.html#typing.ForwardRef), style is only preferred in function stubs when the parameter type import is guarded by if TYPE_CHECKING.

Reply:
But it is, see line 42:
```
    if TYPE_CHECKING:
```
The Queue import is guarded.
In this case we could have used:
```
    _record_q: queue.Queue["Record"]
```
But I think that is harder to understand.  For types I was thinking we should have non package referenced names like:
"Queue", not queue.Queue

For any other functions and classes I think we should always use the package name.   This was a google style requirement and I think we should enforce it.  So none of this:

```
from this.that.anther import this_function
this_function()
```

instead:
```
from this.that import another
another.this_function()
```

I found a flake8 rule to enforce that but havent put it in place yet, and would have to modify much of the code.

Reply:
But if we were using python 3.8 and later we could use the future annotation style where we dont have to stringify this stuff.

Reply:
I would personally prefer `queue.Queue["Record"]` over `"Queue[Record]"` - but admit that might just be a personal preference. Particularly because it is more specific and avoids the edge case of two competing names from different packages. 

For the import comment - this is really interesting - basically you always import a module and never a class/function/variable. I like that paradigm, but don't have strong preferences. Happy to conform (: